Client/Application/AppLaunchUtil.swift

```diff
@@ -83,27 +83,26 @@ class AppLaunchUtil {
 
     func setUpPostLaunchDependencies() {
 
-        let persistedCurrentVersion = InstallType.persistedCurrentVersion()
-        let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
-        // upgrade install - Intro screen shown & persisted current version does not match
-        if introScreen != nil && persistedCurrentVersion != AppInfo.appVersion {
-            InstallType.set(type: .upgrade)
-            InstallType.updateCurrentVersion(version: AppInfo.appVersion)
-        }
-
+        // Ecosia: Removing unneded `introScreen` constant
+        // let introScreen = profile.prefs.intForKey(PrefsKeys.IntroSeen)
+        
+        // Ecosia: InstallType logic review
+        
         // We need to check if the app is a clean install to use for
-        // preventing the What's New URL from appearing.
-        if introScreen == nil {
-            // fresh install - Intro screen not yet shown
+        // preventing the What's New Page from appearing.
```

Comment:
So the only place that will use this is our own What's New Page? Asking to be on the safe side when updating FFs code since it might break something else.

Also, if it will be only us using it, I would argue it makes more sense to threat it as our own code by copying into our Ecosia folder (keeping their version as they have). Even if we re-use the logic this would prevent any conflicts with them if indeed is something only used by us and I'm not getting that wrong 🙈

Reply:
It is also used at `DefaultBrowserOnboardingViewModel:72`, which makes it even more sense to having it with the current logic in place.

> I would argue it makes more sense to threat it as our own code by copying into our Ecosia folder (keeping their version as they have)
If that has been seen as to be a good practice, I'm up for the suggestion. Don't have any strong opinion on following that instead of the current changes.

Are you suggesting to essentially leave everything as is here and extract the logic that we have now into another function that will be called in the same AppDelegate event?

Something like (even with different keys 🤔):

```swift
    func setEcosiaVersionInstallType() {
        if InstallType.get() == .unknown {
            InstallType.set(type: .fresh)
            InstallType.updateCurrentVersion(version: AppInfo.ecosiaAppVersion)
            profile.prefs.setString(AppInfo.ecosiaAppVersion, forKey: LatestAppVersionProfileKey)
        } else if profile.prefs.boolForKey(PrefsKeys.KeySecondRun) == nil {
            profile.prefs.setBool(true, forKey: PrefsKeys.KeySecondRun)
        }

        if InstallType.persistedCurrentVersion() != AppInfo.ecosiaAppVersion {
            InstallType.set(type: .upgrade)
            InstallType.updateCurrentVersion(version: AppInfo.ecosiaAppVersion)
        }
    }
```

Not sure whether it may make it an overkill given the domain we have now of what happens in the `BrowserViewController`, knowing how the app behaves and what's expected.

Reply:
Something like that is what I was suggesting yeah, with different keys and even a different `InstallType` logic/component, so there are really no shared logic with FF on that. I know this might be an overkill and even goes against the initial thought of reusing code for simplicity, I just don't really like how it seems tangled with FF code that we have no control if changes. I would not feel very confident of this bringing unintended consequences then 🙈

In the end this is just my initial though while looking at the code and I don't mean to give this simple change additional complexity, I trust your decision if you prefer to reuse it and update it like that 🙂 

You're also right about it being used by FFs default browser onboarding, would be even curious if we still use that component or not, quickly looking at the code seems this can still be presented through some deeplink 🤔